nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -365,9 +388,12 @@ public void process(InputStream in) throws IOException {
                     final long millis = stopWatch.getDuration(TimeUnit.MILLISECONDS);
                     tempDotCopyFile = tempCopyFile;
 
-                    if (!conflictResponse.equals(APPEND_RESOLUTION)
-                            || (conflictResponse.equals(APPEND_RESOLUTION) && !destinationExists)) {
+                    if  (
+                        writingStrategy.equals(WRITE_AND_RENAME)
```

Comment:
This can be simplified to:
```suggestion
                        writingStrategy.equals(WRITE_AND_RENAME) && (!conflictResponse.equals(APPEND_RESOLUTION) || !destinationExists)
```